CodeEdit/Features/Tasks/CEActiveTask.swift

```diff
@@ -0,0 +1,229 @@
+//
+//  CEActiveTask.swift
+//  CodeEditTests
+//
+//  Created by Tommy Ludwig on 24.06.24.
+//
+
+import SwiftUI
+import Combine
+
+/// Stores the state of a task once it's executed
+class CEActiveTask: ObservableObject, Identifiable, Hashable {
+    /// The current progress of the task.
+    @Published private(set) var output: String  = ""
+
+    /// The status of the task.
+    @Published private(set) var status: CETaskStatus = .notRunning
+
+    /// The name of the associated task.
+    @ObservedObject var task: CETask
+
+    var process: Process?
+    var outputPipe: Pipe?
+
+    private var cancellables = Set<AnyCancellable>()
+
+    init(task: CETask) {
+        self.task = task
+        self.process = Process()
+        self.outputPipe = Pipe()
+
+        self.task.objectWillChange.sink { _ in
+            self.objectWillChange.send()
+        }.store(in: &cancellables)
+    }
+
+    func run() {
+        guard let process, let outputPipe else { return }
+
+        Task { await updateTaskStatus(to: .running) }
```

Comment:
This entire function should be wrapped in a detached task so this await is waited for. Otherwise there's a race condition here where the task could finish before the status is updated.

Alternatively it could be required that this is called from the main thread, and the status is updated first thing w/o an `await`.